frontend/src/handlers/createCommunicationTemplateSubmitHandler.ts

```diff
@@ -17,18 +17,24 @@ export async function createCommunicationTemplateSubmitHandler(
     values: CommunicationTemplateEntity,
     router: NextRouter,
     mutate: ScopedMutator
-) {
+): Promise<ICommunicationTemplate> {
     let result: ICommunicationTemplate;
     if (url) {
         result = await editCommunicationTemplate(url, values);
     } else {
         result = await createNewCommunicationTemplate(values);
     }
-    const id = extractIdFromCommunicationTemplateUrl(result._links.self.href);
+    const id = router.query.id;
```

Comment:
Right now you assume in your handler that you are on a certain url. This is not really pretty because the handler is very dependent on the component right now, worse, on the page.